brightnessd.c

```diff
@@ -1045,7 +1059,67 @@ static uint8_t event_loop(struct Tglobalstate *pglobalstate, struct Txcb *pxcb,
 }
 
 ///////////////////////////////////////////////////////////////////////////////
-// parse_uint8_t()
+// spawn_async
+///////////////////////////////////////////////////////////////////////////////
+/** Asynchronously execute a shell command, i.e. don't wait for it to complete.
+
+    @param cmd              the command to be executed in a shell
+*/
+static void spawn_async(char* cmd) {
+    DEBUG("[spawn_async] executing external command: %s\n", cmd);
+    pid_t process = fork();
+    if (process < 0) {
+        ERROR("[spawn_async] fork error");
+    } else if (process == 0) { // child
+        char *argv[64];
+        char* cmd_dup = strdup(cmd);
+        if (parse_cmd(cmd_dup, 64, argv) != 0 || *argv == NULL) {
+            ERROR("[spawn_async] Unable to parse cmd string\n");
+        } else {
+            int ret_code = execvp(*argv, argv);
+            if (ret_code != 0) {
+                ERROR("[spawn_async] external command failed with exit code: %d\n", ret_code);
+            }
+        }
+        free(cmd_dup);
```

Comment:
Here you `free(cmd_dup)` in case `parse_cmd()` failed or `execvp()` failed. If it hadn't failed, the process is replaced, i.e., `cmd_dup` is "`free()`d" it via replacing the process. OK for me.
However, shouldn't the child terminate itself on `execvp()` failing?

Reply:
Yep, tricky but should be correct. I added the exit() call in case the child fails.

Reply:
Hm, doesn't the child process have to `exit()` in either case? Otherwise, the parent and the child, in case of `parse_cmd()` failed, both continue executing the same code paths, unwinding the same stack, right?